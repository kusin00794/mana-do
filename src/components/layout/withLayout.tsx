import React, { FC } from 'react';
import { RouteProps } from 'react-router-dom';

const withLayout = (Layout: FC, Component: FC) => (props: RouteProps) => {
  return (
    <Layout>
      <Component {...props} />
    </Layout>
  );
};

export default withLayout;
