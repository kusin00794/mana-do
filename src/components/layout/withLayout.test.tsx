import React, { FC } from 'react';
import withLayout from './withLayout';
import { testSnapshots } from '../../utils/test';

const MockLayout: FC = ({ children }) => <div>{children}</div>;
const MockComponent: FC = () => <div>Mocked Component</div>;

describe('Snapshot for withLayout', () => {
  testSnapshots(withLayout(MockLayout, MockComponent), [
    {
      props: {},
      description: 'render withLayout',
    },
  ]);
});
