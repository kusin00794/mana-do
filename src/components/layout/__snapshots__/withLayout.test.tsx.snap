// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for withLayout snapshots render withLayout 1`] = `
<div>
  <div>
    Mocked Component
  </div>
</div>
`;
